packit_service/service/api/webhooks.py

```diff
@@ -206,31 +208,94 @@ def post(self):
 
         return "Webhook accepted. We thank you, Gitlab.", HTTPStatus.ACCEPTED
 
-    @staticmethod
-    def validate_token():
+    def create_issue_token(self):
+        project_data = request.json["project"]
+
+        namespace, repo_name = project_data["path_with_namespace"].split("/")
+        http_url = project_data["git_http_url"]
+
+        project_token = ProjectTokenModel.get_project(
+            namespace=namespace, repo_name=repo_name, project_url=http_url
+        )
+
+        if not project_token:
+            token = jwt.encode(
+                {"namespace": namespace, "repo_name": repo_name},
+                "secret",
+                algorithm="HS256",
+            )
+
+            ProjectTokenModel.create(
+                token=token,
+                namespace=namespace,
+                repo_name=repo_name,
+                project_url=http_url,
+            )
+
+            project = config.get_project(url=http_url)
+            packit_user = project.service.user.get_username()
+
+            logger.info("CONFIDENTIAL ISSUE CREATED HERE")
+
+            project.create_issue(
+                title="Packit-Service Authentication",
+                body=f"To configure Packit-Service with `{repo_name}` you need to\n"
+                f"configure the webhook settings. Head to {http_url}/hooks and add\n"
+                f"the SECRET_TOKEN `{token}` to authenticate requests coming to Packit.\n\n"
+                f"Grant [{packit_user}](https://gitlab.com/{packit_user}) `admin` permissions\n"
+                f"on the {namespace}/{repo_name} project. You can add Packit user by clicking\n"
+                f"[here]({http_url}/-/project_members) manually to match the Packit configuration.",
+                private=True,
+            )
+
+    def validate_token(self):
         """
         https://docs.gitlab.com/ee/user/project/integrations/webhooks.html#secret-token
         """
         if "X-Gitlab-Token" not in request.headers:
             if config.validate_webhooks:
                 msg = "X-Gitlab-Token not in request.headers"
-                logger.warning(msg)
-                raise ValidationFailed(msg)
+                logger.info(msg)
+                self.create_issue_token()
```

Comment:
The naming is a bit misleading. Do we create token or issue here?


```suggestion
                self.create_confidential_issue_with_token()
```

Reply:
👍